api/src/test/java/com/qis/api/service/UserServiceTests.java

```diff
@@ -0,0 +1,116 @@
+package com.qis.api.service;
+
+import com.google.gson.Gson;
+import com.qis.api.entity.User;
+
+import java.sql.Timestamp;
+import java.util.Optional;
+
+import com.qis.api.helpers.UserHelper;
+import com.qis.api.repository.UserRepository;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class UserServiceTests {
+
+    private Gson gson = new Gson();
+
+    @InjectMocks
+    private UserService userService;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @BeforeEach
+    public void initMocks() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void findById_WhenUserExists_ReturnsUser() {
+        // Arrange
+        User user = UserHelper.createTestUser();
+        Mockito
+                .when(userRepository.findById(Mockito.anyString()))
+                .thenReturn(Optional.of(user));
+
+        // Act
+        Optional<User> retrievedUser = userService.findById(user.getId());
+
+        // Assert
+        Assertions.assertNotNull(retrievedUser);
+
+        String userAsJson = gson.toJson(Optional.of(user));
+        String retrievedUserAsJson = gson.toJson(retrievedUser);
+
+        Assertions.assertEquals(userAsJson, retrievedUserAsJson);
+    }
+
+    @Test
+    void findById_WhenUserDoesNotExist_ReturnsNull() {
+        // Arrange
+        Mockito
+                .when(userRepository.findById(Mockito.anyString()))
+                .thenReturn(null);
```

Comment:
The actual repository will return an empty optional not null
```suggestion
                .thenReturn(Optional.empty());
```